control/plugin/execution.go

```diff
@@ -115,20 +115,35 @@ func (e *ExecutablePlugin) Run(timeout time.Duration) (Response, error) {
 	e.cmd.Start()
 	e.captureStderr()
 	go func() {
-		for stdOutScanner.Scan() {
-			// The first chunk from the scanner is the plugin's response to the
-			// handshake.  Once we've received that, we can begin to forward
-			// logs on to snapd's log.
-			if !respReceived {
-				respBytes := stdOutScanner.Bytes()
-				err = json.Unmarshal(respBytes, &resp)
-				respReceived = true
-				close(doneChan)
-			} else {
-				execLogger.WithFields(log.Fields{
-					"plugin": path.Base(e.cmd.Path()),
-					"io":     "stdout",
-				}).Debug(stdOutScanner.Text())
+		for {
+			for stdOutScanner.Scan() {
+				// The first chunk from the scanner is the plugin's response to the
+				// handshake.  Once we've received that, we can begin to forward
+				// logs on to snapd's log.
+				if !respReceived {
+					respBytes := stdOutScanner.Bytes()
+					err = json.Unmarshal(respBytes, &resp)
+					respReceived = true
+					close(doneChan)
+				} else {
+					execLogger.WithFields(log.Fields{
+						"plugin": path.Base(e.cmd.Path()),
+						"io":     "stdout",
+					}).Debug(stdOutScanner.Text())
+				}
+			}
+
+			if err := stdOutScanner.Err(); err != nil {
+				reader := bufio.NewReader(e.stdout)
+				log, err := reader.ReadString('\n')
+
+				execLogger.
+					WithField("plugin", path.Base(e.cmd.Path())).
+					WithField("io", "stdout").
+					WithField("scanner_err", stdOutScanner.Err()).
+					WithField("read_string_err", err).
+					Warn(log)
+				continue
```

Comment:
I don't think this continue is needed.

Also there is still a case where we will want to break out of the loop right? Otherwise this goroutine will never exit?